animation_nodes/nodes/rotation/c_utils.pyx

```diff
@@ -100,3 +101,29 @@ def getAxisListOfQuaternionList(QuaternionList quaternions, str axis):
         for i in range(output.length):
             output.data[i] = quaternions.data[i].z
     return output
+
+def randomQuaternionList(int seed, int amount):
+    cdef QuaternionList result = QuaternionList(length = amount)
+    cdef Quaternion *values = <Quaternion*>result.data
+    cdef XoShiRo256Plus rng = XoShiRo256Plus(seed)
+    cdef double length
+    cdef double w, x, y, z
+    cdef Py_ssize_t i
+    for i in range(amount):
+        w = 1.0
+        x = rng.nextDouble()
+        y = rng.nextDouble()
+        z = rng.nextDouble()
+
+        length = sqrt(x * x + y * y + z * z + w * w)
+        w /= length
+        x /= length
+        y /= length
+        z /= length
```

Comment:
You can use `quaternionNormalize_InPlace` here. After setting the values, call `quaternionNormalize_InPlace(values + i)`.